pkg/vgmanager/cleanup.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vgmanager
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/red-hat-storage/lvm-operator/controllers"
+	topolvmv1 "github.com/topolvm/topolvm/api/v1"
+	lvmCMD "github.com/topolvm/topolvm/pkg/lvmd/cmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+)
+
+// vgCleanupAll removes all volume groups existing in lvmd config file and after
+// removing every vg it also removes lvmd config file
+func vgCleanupAll(r *VGReconciler, ctx context.Context) error {
+
+	// get limited number of logical volume CRs to verify their presence
+	lvs := &topolvmv1.LogicalVolumeList{}
+	// TODO: perform cleanup based on individual volume groups after comparing
+	// lvmd config file with lvmcluster CR
+	if err := r.Client.List(ctx, lvs, &client.ListOptions{Limit: int64(1)}); err == nil {
+		// do not proceed with cleanup if there are logicalVolumes
+		if len(lvs.Items) > 0 {
+			return fmt.Errorf("found presence of logicalvolumes and aborting volume group cleanup")
+		}
+	} else {
+		r.Log.Error(err, "failed to list logicalvolumes CRs")
+		return err
+	}
+
+	// get all the volume groups from lvmd file
+	lvmdConfig := &lvmCMD.Config{
+		SocketName: controllers.DefaultLVMdSocket,
+	}
+
+	cfgBytes, err := os.ReadFile(controllers.LvmdConfigFile)
+	if os.IsNotExist(err) {
+		r.Log.Info("lvmd config file doesn't exist, no need of cleanup")
+		return nil
+	} else if err != nil {
+		r.Log.Error(err, "failed to read lvmd config file", "file path", controllers.LvmdConfigFile)
+		return err
+	} else {
+		err = yaml.Unmarshal(cfgBytes, &lvmdConfig)
+		if err != nil {
+			r.Log.Error(err, "failed to unmarshal yaml contents of lvmd config data")
+			return err
+		}
+	}
+
+	// remove volume group sequentially
+	var done int
+	isStarted := false
+	for done = 0; done < len(lvmdConfig.DeviceClasses); done++ {
+		deviceClass := lvmdConfig.DeviceClasses[done]
+		if err = RemoveVolumeGroup(deviceClass.Name); err != nil {
+			if errors.Is(err, ErrNotFound) {
```

Comment:
It would be nice to log the error for other cases than "not found", the FindVolumeGroup function can fail executing a command and we will lost the information of what happened.